client/Components/Decks/DeckEditor.jsx

```diff
@@ -1,54 +1,92 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import _ from 'underscore';
-import $ from 'jquery';
 import { connect } from 'react-redux';
 
 import Input from '../Form/Input';
 import Select from '../Form/Select';
 import Typeahead from '../Form/Typeahead';
 import TextArea from '../Form/TextArea';
+import validateDeck from '../../deck-validator';
 import * as actions from '../../actions';
 
 class DeckEditor extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
+            bannerCards: [],
             cardList: '',
-            deck: this.copyDeck(props.deck),
+            deckName: 'New Deck',
+            drawCards: [],
+            faction: props.factions && props.factions['baratheon'],
             numberToAdd: 1,
-            showBanners: props.deck.agenda && props.deck.agenda.code === '06018',
+            plotCards: [],
+            showBanners: false,
             selectedBanner: {},
             validation: {
                 deckname: '',
                 cardToAdd: ''
             }
         };
-    }
 
-    componentWillMount() {
-        if(!this.props.deck.faction && this.props.factions) {
-            let deck = this.copyDeck(this.state.deck);
+        if(props.deck) {
+            this.state.deckId = props.deck._id;
+            this.state.deckName = props.deck.name;
+            this.state.plotCards = props.deck.plotCards;
+            this.state.drawCards = props.deck.drawCards;
+            this.state.bannerCards = props.deck.bannerCards;
```

Comment:
Should copies of these arrays be made instead of direct references? From the looks of the code flow, this will result in the deck in the store being updated as well, so I'm not 100% confident it fixes #1825.

Reply:
The deck is now owned by the add/edit components and is not in redux at all, so it definitely fixes it :)